test/jdk/sun/security/validator/CertReplace.java

```diff
@@ -93,58 +85,61 @@ private static void certReplace() throws Exception {
 
         // 1. Generate 3 aliases in a keystore: ca, int, user
         SecurityTools.keytool(ktBaseParameters +
-                              "-genkeypair -alias " + caAlias + " -dname CN=CA -keyalg rsa -sigalg md2withrsa -ext bc");
+                              "-genkeypair -alias ca -dname CN=CA -keyalg rsa -sigalg md2withrsa -ext bc");
         SecurityTools.keytool(ktBaseParameters +
-                              "-genkeypair -alias " + intAliase + " -dname CN=Int -keyalg rsa");
+                              "-genkeypair -alias int -dname CN=Int -keyalg rsa");
         SecurityTools.keytool(ktBaseParameters +
-                              "-genkeypair -alias " + userAlias + " -dname CN=User -keyalg rsa");
+                              "-genkeypair -alias user -dname CN=User -keyalg rsa");
 
         final KeyStore keyStore = KeyStoreUtils.loadKeyStore(CERTREPLACE_JKS, PASSWORD);
 
         // 2. Signing: ca -> int -> user
-        final String intReqFile = intAliase + ".req";
-        final String intCertFileName = intAliase + ".cert";
 
         SecurityTools.keytool(ktBaseParameters +
-                              "-certreq -alias " + intAliase + " -file " + intReqFile);
+                              "-certreq -alias int -file int.req");
         SecurityTools.keytool(ktBaseParameters +
-                              "-gencert -rfc -alias " + caAlias + " -ext bc -infile " + intReqFile + " " +
-                              "-outfile " + intCertFileName);
+                              "-gencert -rfc -alias ca -ext bc -infile int.req " +
+                              "-outfile int.cert");
 
         //putting the certificate in the keystore
         final CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+
+        final FileInputStream certInputStream = new FileInputStream("int.cert");
         final Certificate[] certs = new Certificate[]{
-                certificateFactory.generateCertificate(
-                        new FileInputStream(intCertFileName)
+                CertUtils.getCertFromStream(
+                        certInputStream
                 )
         };
-        final PrivateKey privateKey = (PrivateKey) keyStore.getKey(intAliase, PASSWORD.toCharArray());
-        keyStore.setKeyEntry(intAliase, privateKey, PASSWORD.toCharArray(), certs);
-        keyStore.store(new FileOutputStream(CERTREPLACE_JKS), PASSWORD.toCharArray());
+        certInputStream.close();
```

Comment:
Usually we use `try (FileInputStream...)` but your style is also OK. Otherwise we will need to define `certs` outside. You choose.